selectrum.el

```diff
@@ -1512,10 +1504,11 @@ shadows correctly."
                  (cl-return)))
              (cl-incf num-components)))))
      table)
-    (get-text-property
-     0 'selectrum--lib-path
-     (selectrum-read
-      "Library name: " lst :require-match t :may-modify-candidates t))))
+    (let ((res (selectrum-read
+                "Library name: " lst
+                :require-match t :may-modify-candidates t)))
+      (get-text-property
+       0 'selectrum--lib-path (car (member res lst))))))
```

Comment:
But, in the case of duplicates (this will happen when there are `load-path` shadows), how can you be sure of picking the right one? That's the original reason that I wanted to use text properties. If, indeed, it is part of the `completing-read` API to strip text properties, I am not sure what the right thing to do is... but I am pretty sure we need to handle duplicates properly, at least for this case.

Perhaps there could be a dynamically bound variable, non-nil by default, for whether to strip text properties.

Reply:
Oh, right I did not thought about load path shadows. I just checked and the regular  `completing-read` does remove any duplicates in the collection so that is just a case `completing-read` does not handle. 

> Perhaps there could be a dynamically bound variable, non-nil by default, for whether to strip text properties.

Wouldn't it be better to make that an `selectrum-read` argument? Because relying on that dynamic variable would break when `completing-read` is used without selectrum.

Reply:
I have pushed a commit which includes the prefix in the candidate name. This way there are no dups.

Reply:
I don't really like this approach mainly because the sorting and filtering makes less sense to me. In `selectrum-prescient.el`, for example, it is assumed that shorter candidates should be put at the front. But with this change, any libraries that have load-path shadows will seem like they have long names, and will be sorted last. Also, the feature where typing in `org` will force each copy of the `org` library to the front of the list (i.e. exact matching) no longer works.

I also suspect that others will find it useful for text properties to be preserved. How would the Swiper-like command from the wiki be ported to this version of Selectrum? I don't think adding line numbers directly to the candidates would work very well -- it would make it confusing to search for numbers, for example, and it's in conflict with how other tools work, such as Swiper itself.

But now that I'm thinking, what's wrong with just making `selectrum-completing-read` strip text properties from the candidate returned by `selectrum-read`? Wouldn't that solve this problem without really any downside?

Reply:
> I also suspect that others will find it useful for text properties to be preserved. How would the Swiper-like command from the wiki be ported to this version of Selectrum? I don't think adding line numbers directly to the candidates would work very well -- it would make it confusing to search for numbers, for example, and it's in conflict with how other tools work, such as Swiper itself.

The command uses `completing-read`, following the advice of the wiki. Since `completing-read` strips the text properties, I did change it to include the line numbers in the candidate. 

You are correct that it is worse for searching for numbers (which I didn't consider), but it works well enough otherwise. Is it common it is to search for numbers? My experience is that important numbers are usually given a name.

Adding the line numbers also helps to make the candidates distinct. By doing it this way, it also works in Icomplete and the regular `*Completions*` buffer. Currently, the command manually removes the line numbers from its history variable, since I think that is more expected.

Please try it and let me know what you think. It currently satisfies my needs, but it is better to be useful for many.

Thank you.

Reply:
@raxod502 

Good points, I reverted the commit and now strip the text properties only for the built-in completion functions.